src/interface/screenshot.c

```diff
@@ -80,19 +81,43 @@ static int screenshot_get_next_path(char *path)
 		return -1;
 	}
 
-	int i;
-	for (i = 1; i < 1000; i++) {
-		set_format_arg(0, uint16, i);
+	char park_name[128] = { 0 };
+	format_string(park_name, gParkName, &gParkNameArgs);
 
-		// Glue together path and filename
-		sprintf(path, "%sSCR%d.png", screenshotPath, i);
+	// retrieve current time
+	time_t now;
+	time(&now);
+	struct tm* tm_now = localtime(&now);
+
+	utf8 time_now_str[128] = { 0 };
+	size_t size = strftime(time_now_str, 128, "%F %H-%M-%S", tm_now);
+	assert(size != 0);
 
+	sprintf(path, "%s%s %s.png", screenshotPath, park_name, time_now_str);
+
+	if (!platform_file_exists(path)) {
+		return 0; // path ok
+	}
+
+	// multiple screenshots with same timestamp
+	// might be possible when switching timezones
+	// in the unlikely case that this does happen,
+	// append (%d) to the filename and increment
+	// this int until it doesn't overwrite any
+	// other file in the directory.
+	int i = 2;
+	do {
+		set_format_arg(0, uint16, i);
+		// Glue together path and filename
+		sprintf(path, "%s%s %s (%d).png", screenshotPath, park_name, time_now_str, i);
+		
 		if (!platform_file_exists(path)) {
 			return i;
 		}
-	}
+		i++;
+	} while (platform_file_exists(path));
```

Comment:
Before it Before there was a limit of 1000, now `i` will keep incrementing until it overflows and it will get stuck. Unlikely I know but... the code below will never get hit otherwise.